src/test/java/com/snowflake/kafka/connector/internal/ingestsdk/StreamingClientManagerTest.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2023 Snowflake Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.snowflake.kafka.connector.internal.ingestsdk;
+
+import com.snowflake.kafka.connector.internal.SnowflakeErrors;
+import com.snowflake.kafka.connector.internal.TestUtils;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+public class StreamingClientManagerTest {
+
+  @Test
+  public void testCreateAndGetAllStreamingClients() {
+    // test to create the following mapping
+    // [0, 1] -> clientA, [2, 3] -> clientB, [4] -> clientC
+    // test
+    StreamingClientManager manager = new StreamingClientManager();
+    manager.createAllStreamingClients(TestUtils.getConfForStreaming(), "testkcid", 5, 2);
+
+    // verify
+    KcStreamingIngestClient task0Client = manager.getValidClient(0);
+    KcStreamingIngestClient task1Client = manager.getValidClient(1);
+    assert task0Client.equals(task1Client);
+
+    KcStreamingIngestClient task2Client = manager.getValidClient(2);
+    KcStreamingIngestClient task3Client = manager.getValidClient(3);
+    assert task2Client.equals(task3Client);
+    assert !task2Client.equals(task0Client);
+
+    KcStreamingIngestClient task4Client = manager.getValidClient(4);
+    assert !task4Client.equals(task0Client);
+    assert !task4Client.equals(task2Client);
```

Comment:
one edge case I can think of is:
lets say 6 partitions
6 tasks
2 connector instances

So each connector instance will have 3 tasks spun up, will they have 0,1,2 and 0,1,2 or 0,1,2 and 3,4,5. 
[I am guessing 0,1,2 and 3,4,5, but want to confirm if this is what you expect too]

Reply:
that is what is expected, i'll add a test for an even divide like your example just to confirm